### PR TITLE
feat: adding filter infrastructure + status and event filter

### PR DESCRIPTION
**Pull request by CodeChanning**
Implementing `--filter` flag for `nerdctl events`, Addresses part of #54

This change adds infrastructure for adding filters. Similar to Podman's implementation of `--filter`.
 
I also added Status and Event filter and unit tests for those. 

**Context:**
- Listed as unimplemented in [nerdctl command reference](https://github.com/containerd/nerdctl/blob/main/docs/command-reference.md#whale-nerdctl-events)
- [Podman's implementation](https://github.com/containers/podman/blob/189d862d54b3824c74bf7474ddfed6de69ec5a09/libpod/events/filters.go#L11)

### CODE REVIEW

**Comment by CodeChanning** [comment]
@AkihiroSuda PTAL when you get the chance

File: docs/command-reference.md
Comment (yankay) [inline docs/command-reference.md:1277]: HI @CodeChanning 

The new flag docs can be added here. :-)

File: pkg/cmd/system/events.go
Comment (yankay) [inline pkg/cmd/system/events.go:80]: In Golang, usually, the comments should start with the Function Name. When documenting a function, the first sentence should typically start with the function's name. This format helps quickly clarify the function’s purpose.

File: docs/command-reference.md
Comment (CodeChanning) [reply docs/command-reference.md:1277]: Added `events --filter` docs. PTAL and let me know if this is sufficient.

File: pkg/cmd/system/events.go
Comment (CodeChanning) [reply pkg/cmd/system/events.go]: I changed the comments. Let me know if that works.

**Comment by Shubhranshu153** [comment]
@AkihiroSuda 
The --sig-proxy, -a support in run and the event pr (this one) we have been adding to support dev containers on finch. Considering these are not breaking changes will it. be possible to back port these pr's to 1.7 ... or we don't plan to release any new things in 1.7 label.

File: cmd/nerdctl/system_events_linux_test.go
Comment (AkihiroSuda) [inline cmd/nerdctl/system_events_linux_test.go:101]: Could you add a comment to explain the reason?

**Comment by AkihiroSuda** [comment]
> 1.7

I guess this is safe to cherry-pick. Feel free to open cherry-pick PRs after merging to the main branch.
cc @containerd/nerdctl-committers

File: cmd/nerdctl/system_events_linux_test.go
Comment (AkihiroSuda) [reply cmd/nerdctl/system_events_linux_test.go:101]: ping @CodeChanning

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**
